src/containers/user/UserSettings.js

```diff
@@ -0,0 +1,227 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { prefetch } from 'react-fetcher'
+import cx from 'classnames'
+const { func, object } = React.PropTypes
+import { fetchCurrentUser, updateUserSettings, updateUserSettingsEditor } from '../../actions'
+
+@prefetch(({ dispatch, params: {id} }) => dispatch(fetchCurrentUser()))
+@connect(({ people }, props) => ({currentUser: people.current}))
+export default class UserSettings extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {editing: {}, edited: {}, errors: {}}
+  }
+
+  static propTypes = {
+    currentUser: object,
+    dispatch: func,
+    location: object
+  }
+
+  updateStore (data) {
+    let { dispatch } = this.props
+    dispatch(updateUserSettingsEditor(data))
+  }
+
+  validate () {
+    let { errors } = this.state
+
+    if (errors.email) {
+      window.alert('Please provide a valid email.')
+      this.refs.email.focus()
+      return
+    }
+
+    if (errors.password) {
+      window.alert('Please provide a valid password.')
+      this.refs.password.focus()
+      return
+    }
+
+    return true
+  }
+
+  setEmail = event => {
+    return this.setState({
+      edited: {...this.state.edited, email: event.target.value},
+      errors: {...this.state.errors, email: !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(event.target.value)}
+    })
+  }
+
+  setPassword = event => {
+    return this.setState({
+      edited: {...this.state.edited, password: event.target.value},
+      errors: {...this.state.errors, password: !event.target.value}
+    })
+  }
+
+  save = (field) => {
+    if (!this.validate()) return
+
+    let { dispatch, currentUser } = this.props
+    let { editing, edited } = this.state
+    this.setState({editing: {...editing, [field]: false}})
+    dispatch(updateUserSettingsEditor({...currentUser, ...edited}))
+    dispatch(updateUserSettings({...currentUser, ...edited}, {[field]: currentUser[field]}))
+  }
+
+  edit (field) {
+    let { currentUser } = this.props
+    let { editing, edited } = this.state
+    edited[field] = currentUser[field]
+    this.setState({editing: {...editing, [field]: true}})
+  }
+
+  cancelEdit (field) {
+    let { editing } = this.state
+    this.setState({editing: {...editing, [field]: false}})
+  }
+
+  toggle (field) {
```

Comment:
the best refactoring i see would be something like:

```js
update (field, value) {
  let updatedUser = {...currentUser, [field]: value}
  dispatch(...)
}

updateSetting (field, value) (
  let updatedSettings = makeUpdatedSettings(field, value)
  update('settings', updatedSettings)
}
```

...and then just handle the toggle cases by writing out `update('foo', !currentUser.foo)` in the template. this can be pretty concise with liberal use of `let { ... } = currentUser` and `let { ... } = settings`. or write two very small toggle methods that wrap the methods above.

btw, note redundant syntax: 
`{...currentUser, ...{[field]: value}}` =>
`{...currentUser, [field]: value}`